src/AutoRest.CSharp/Output/Models/Types/ParentDetection.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Input;
+
+namespace AutoRest.CSharp.Output.Models.Type.Decorate
+{
+    internal static class ParentDetection
+    {
+        public static string GetParent(OperationGroup operationGroup)
+        {
+            var method = GetBestMethod(operationGroup.OperationHttpMethodMapping);
+            if (method == null)
+            {
+                throw new ArgumentException($"Could not set parent for operations group {operationGroup.Key}. Please add to readme.md");
+            }
+
+            var fullProvider = GetFullProvider(method.ProviderSegments);
+            if (fullProvider == null)
+            {
+                throw new ArgumentException($"Could not set parent for operations group {operationGroup.Key}. Please add to readme.md");
+            }
+            var canidateParent = ParseMethodForParent(fullProvider, method.Path, operationGroup.ResourceType);
+            if (canidateParent == string.Empty)
+            {
+                throw new ArgumentException($"Could not set parent for operations group {operationGroup.Key}. Please add to readme.md");
+            }
+            return canidateParent;
+        }
+
+        public static HttpRequest? GetBestMethod(Dictionary<HttpMethod, List<ServiceRequest>> operations)
+        {
+            List<ServiceRequest>? requests;
+
+            if (operations.TryGetValue(HttpMethod.Put, out requests))
```

Comment:
Add a new TestProject with a swagger that has an operationgroup that hits each of these cases and assert the predetermined parent is chosen